kratos/utilities/string_utilities.cpp

```diff
@@ -175,4 +175,55 @@ std::string ReplaceAllSubstrings(
     return output_string;
 }
 
+std::string Trim(
+    const std::string& rInputString,
+    const bool RemoveNullChar)
+{
+    return TrimLeft(TrimRight(rInputString, RemoveNullChar), RemoveNullChar);
+}
+
+std::function<bool(std::string::value_type)> TrimChar(const bool RemoveNullChar)
```

Comment:
I know this isn't used in performance-critical places, but having `std::function`s for this job just rubs me the wrong way. Would you consider moving the `RemoveNullChar` argument to a template parameter, so that this function could return a lambda instead of an `std::function`?

again: this isn't too important, but if you have nothing against it you could put my mind at ease.

Reply:
I improved the code, what do you think about it now?

Reply:
seems like the current version does not compile in win :facepalm:

Reply:
Ping @matekelemen